src/Maestro/Maestro.Web/DarcRemoteFactory.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Maestro.Data;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.Kusto;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Maestro.Web
+{
+    public class DarcRemoteFactory : IRemoteFactory
+    {
+        public DarcRemoteFactory(
+            BuildAssetRegistryContext context,
+            IKustoClientProvider kustoClientProvider)
+        {
+            Context = context;
+            KustoClientProvider = (KustoClientProvider) kustoClientProvider;
+        }
+        
+        public BuildAssetRegistryContext Context { get; }
+
+        private readonly KustoClientProvider KustoClientProvider;
+
+        public Task<IRemote> GetBarOnlyRemoteAsync(ILogger logger)
+        {
+            return Task.FromResult((IRemote)new Remote(null, new MaestroBarClient(Context, KustoClientProvider), logger));
+        }
```

Comment:
This _feels_ like the caller should just be using DI.  This "I have a factory and I called a GetSomething seems like we are trying to reimplement the dependency injection.

Reply:
They are both required functions to extend IRemote. Should I just make the second not implemented since we do not use it?

Reply:
Bleh. It seems like we are building on a not great design, but it's probably a bit much to redo all that design here. Just leave it.